typing/ikinds/ldd_hybrid_opt.ml

```diff
@@ -133,90 +147,115 @@ module Make (V : ORDERED) = struct
           v)
   end
 
-  (* Subtract subsets h - l (no memoization). *)
-  let rec canonicalize (h : node) (l : node) : node =
-    if h == l
+  (* Subtract subsets hi - lo.
+     This preserves ordering and maintains the canonical form [hi = hi - lo]. *)
+  (* Ordering fast path: if [lo] has a larger top var, it cannot appear
+     under [hi], so we only recurse into [hi]'s subtrees. *)
+  let rec canonicalize ~(hi : node) ~(lo : node) : node =
+    if hi == lo
```

Comment:
```suggestion
  (** Subtract subsets hi - lo.
      This preserves ordering and maintains the canonical form [hi = hi - lo]. *)
  let rec canonicalize ~(hi : node) ~(lo : node) : node =
    (* Ordering fast path: if [lo] has a larger top var, it cannot appear
       under [hi], so we only recurse into [hi]'s subtrees. *)
    if hi == lo
```
I have two motivations for this change:
1. The first comment is basically a doc comment, so I'd like it to get treated as one and get attached to the function properly. It can't do that if there's another comment between it and the function definition.
2. The second comment is about the implementation, which I like to put inside functions. But this is a somewhat arbitrary preference on my part.